airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py

```diff
@@ -426,10 +426,15 @@ def test_export_stream(requests_mock, export_response):
 
     stream = Export(authenticator=MagicMock())
     requests_mock.register_uri("GET", get_url_to_mock(stream), export_response)
-
+    stream_state = {"date" : "2021-06-16T17:00:00"}
     stream_slice = {"start_date": "2017-01-25T00:00:00Z", "end_date": "2017-02-25T00:00:00Z"}
     # read records for single slice
     records = stream.read_records(sync_mode=SyncMode.incremental, stream_slice=stream_slice)
 
     records_length = sum(1 for _ in records)
     assert records_length == 1
+
+    # no records should be returned when state older than the record
+    records = stream.read_records(sync_mode=SyncMode.incremental, stream_slice=stream_slice, stream_state=stream_state)
+    records_length = sum(1 for _ in records)
+    assert records_length == 0
```

Comment:
I don't think this is going to work with unit test and also this is not following the best practices. Please create another test to validate what you're implementing.

Reply:
Yes, forgot to push changed tests after the implementation was modified (v1 was doing client side filtering)